tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -918,11 +924,66 @@ public function it_should_create_mime_types_for_allowed_sizes_only_via_global_va
 		$this->assertImageNotHasSizeSource( $attachment_id, 'not_allowed_size_200x150', 'image/webp' );
 	}
 
+	/**
+	 * Test image quality for image conversion.
+	 *
+	 * @test
+	 */
+	public function it_should_set_quality_with_image_conversion() {
+		$editor = wp_get_image_editor( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/dice.png', array( 'mime_type' => 'image/png' ) );
+
+		// Quality setting for the source image. For PNG the fallback default of 82 is used.
+		$this->assertSame( 82, $editor->get_quality(), 'Default quality setting for PNG is 82.' );
+
+		// Temporary file.
+		$file = wp_tempnam();
+
+		// A PNG image will be converted to WebP whose quality should be 82 universally.
+		$editor->save( $file, 'image/webp' );
+		$this->assertSame( 82, $editor->get_quality(), 'Output image format is WebP. Quality setting for it should be 82 universally.' );
+
+		$editor = wp_get_image_editor( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+
+		// Quality setting for the source image. For JPG the fallback default of 82 is used.
+		$this->assertSame( 82, $editor->get_quality(), 'Default quality setting for JPG is 82.' );
+
+		// A JPG image will be converted to WebP whose quality should be 82 universally.
+		$editor->save( $file, 'image/webp' );
+		$this->assertSame( 82, $editor->get_quality(), 'Output image format is WebP. Quality setting for it should be 82 universally.' );
+
+		// Delete the temporary file.
+		unlink( $file );
+	}
+
+	/**
+	 * Test webp_uploads_modify_webp_quality function for image quality.
+	 *
+	 * @covers ::webp_uploads_modify_webp_quality()
+	 *
+	 * @test
+	 */
+	public function it_should_return_correct_quality_for_mime_types() {
+		$this->assertSame( 82, webp_uploads_modify_webp_quality( 90, 'image/webp' ), 'WebP image quality should always be 82.' );
+		$this->assertSame( 82, webp_uploads_modify_webp_quality( 82, 'image/webp' ), 'WebP image quality should always be 82.' );
+		$this->assertSame( 80, webp_uploads_modify_webp_quality( 80, 'image/jpeg' ), 'JPEG image quality should return default quality provided from WP filter wp_editor_set_quality.' );
+		$this->assertNotSame( 50, webp_uploads_modify_webp_quality( 40, 'image/jpeg' ), 'JPEG image quality should return default quality provided from WP filter wp_editor_set_quality.' );
+	}
+
 	/**
 	 * Runs (empty) hooks to satisfy webp_uploads_in_frontend_body() conditions.
 	 */
 	private function mock_frontend_body_hooks() {
 		remove_all_actions( 'template_redirect' );
 		do_action( 'template_redirect' );
 	}
+
+	/**
+	 * Return WebP image quality 86 for testing.
+	 */
+	public function return_webp_image_quality( $quality, $mime_type ) {
+		if ( 'image/webp' === $mime_type ) {
+			return 86;
```

Comment:
Maybe use something even higher, to make it even more safe that the created WebP image is larger than the JPEG. How about `90`?

Reply:
Per https://github.com/WordPress/performance/pull/571#discussion_r1019633576 how about we also filter the quality of jpg (with another method) to ensure webp will always be larger?

Reply:
See my comment above for this ^